pkg/reconciler/v1alpha1/taskrun/resources/output_resource.go

```diff
@@ -186,3 +190,13 @@ func addStoreUploadStep(build *buildv1alpha1.Build,
 	build.Spec.Steps = append(build.Spec.Steps, buildSteps...)
 	return nil
 }
+
+// allowedOutputResource checks if an output resource type produces
+// an output that should be copied to the PVC
+func allowedOutputResource(resourceType v1alpha1.PipelineResourceType) bool {
```

Comment:
`allowedOutputResources` is not going to change so makes sense to define it in variable section. Something like 
```
var (
// allowedOutputResource checks if an output resource type produces
// an output that should be copied to the PVC
	allowedOutputResources = map[v1alpha1.PipelineResourceType]bool{
		v1alpha1.PipelineResourceTypeStorage: true,
		v1alpha1.PipelineResourceTypeGit:     true,
	}
)
```

and update the if condition to be 
```
if allowedOutputResources[resource.Spec.Type] && pipelineRunpvcName != "" {
}
```

Code is simple and does not need a function. WDYT?

Reply:
I made a function because you asked for it to begin with :)

I'm not sure that list is not going to change... other resources will have output to be copied, like logs of the build process for the image resource.
I hope eventually that list will go away for good and it will be replaced by a proper interface on the resource to declare whether they have output to be copied or not. 

The advantage of having a function is also that it's possible to write a simple unit test for it, but it's true that there's little value in that unit test, so I'm ok with dropping the function if you prefer so.

Reply:
> I made a function because you asked for it to begin with :)

Sorry for the confusion. My intention was to make the `if` condition simpler and easier to read so I jumped at function as solution. I should have left up to you come up with simpler solution instead of influencing my solution on you. After I saw your solution it felt declaring map in variables section seemed a lot simpler than function

Reply:
> The advantage of having a function is also that it's possible to write a simple unit test for it, but it's true that there's little value in that unit test, so I'm ok with dropping the function if you prefer so.

You can still have the tests. I think it holds value in showing supported output types explicitly. I like the tests. I have suggested below how to update tests to use map instead of function.

Reply:
OK, addressed everything - hopefully - in the version :)